product_life_period/views/product_life_period_views.xml

```diff
@@ -0,0 +1,162 @@
+            <field name="name">product.life.period.search</field>
+            <field name="model">product.life.period</field>
+            <field name="arch" type="xml">
+                <search string="Product Life Period">
+                    <field name="name"/>
+                    <filter string="Inactive" name="inactive" domain="[('active','=',False)]"/>
+                </search>
+            </field>
+        </record>
+
+        <record id="product_life_period_view_tree" model="ir.ui.view">
+            <field name="name">product.life.period.tree</field>
+            <field name="model">product.life.period</field>
+            <field name="arch" type="xml">
+                <tree string="Product Life Period">
+                    <field name="name"/>
+                    <field name="start_date"/>
+                    <field name="end_date"/>
+                </tree>
+            </field>
+        </record>
+
+        <record id="action_open_life_period_products" model="ir.actions.act_window">
+            <field name="name">Product Life Period</field>
+            <field name="type">ir.actions.act_window</field>
+            <field name="res_model">product.template</field>
+            <field name="view_mode">kanban,form,tree</field>
+            <field name="view_type">form</field>
+            <field name="domain">[('product_life_period_id', '=', active_id)]</field>
+        </record>
+
+        <record id="product_life_period_action" model="ir.actions.act_window">
+            <field name="name">Product Life Period</field>
+            <field name="type">ir.actions.act_window</field>
+            <field name="res_model">product.life.period</field>
+            <field name="view_mode">tree,form</field>
+            <field name="view_type">form</field>
+        </record>
+
+        <record id="product_life_period_view_form" model="ir.ui.view">
+            <field name="name">product.life.period.form</field>
+            <field name="model">product.life.period</field>
+            <field name="arch" type="xml">
+                <form string="Product Life Period">
+                    <sheet>
+                        <div class="oe_title">
+                            <div class="oe_edit_only">
+                                <label for="name"/>
+                            </div>
+                            <h1>
+                                <field name="name"/>
+                            </h1>
+                        </div>
+                        <div class="oe_right oe_button_box">
+                            <button
+                                class="oe_inline oe_stat_button"
+                                type="action"
+                                name="%(action_open_life_period_products)d"
+                                icon="fa-cubes">
+                                <field name="products_count" string="Products" widget="statinfo" />
```

Comment:
It should be great to pass the current period in context as default value on product action